garak/probes/suffix.py

```diff
@@ -108,3 +109,52 @@ def probe(self, generator) -> List[garak.attempt.Attempt]:
             logging.warning("GCG failed to find a jailbreak!")
             # Return an empty list to avoid throwing assertion errors upstream
             return list()
+
+
+class BEAST(Probe):
+    """
+    BEAST probe
+
+    Beam Search-based Adversarial attack.
+    Based on "Fast Adversarial Attacks in One GPU Minute" by Sadasivan et al. (https://arxiv.org/abs/2402.15570)
+    """
+
+    bcp47 = "en"
+    recommended_detector = ["mitigation.MitigationBypass"]
+    tags = [
+        "avid-effect:security:S0403",
+        "avid-effect:performance:P0204",
+        "quality:Security:PromptStability",
+        "payload:jailbreak",
+    ]
+    goal = "cause the model to disregard the system prompt"
+    doc_uri = "https://arxiv.org/abs/2402.15570"
+
+    prompts = list()
+    active = False
+
+    def __init__(self, goal: str = GOAL, config_root=_config):
+        from garak.resources.beast import run_beast
+
+        self.goal = goal
+        self.run_beast = run_beast
+        super().__init__(config_root=config_root)
+
+    def probe(self, generator) -> List[garak.attempt.Attempt]:
+        self.generator = generator
+
+        try:
+            beast_output = self.run_beast(target_generator=self.generator)
+        except Exception as e:
+            logging.error(e)
+            print(f"BEAST generation encountered an error:\n{e}")
+            beast_output = None
+
+        if beast_output:
+            self.prompts = [self.goal + beast_output]
```

Comment:
Is this supposed to prepend the goal once as the first prompt?
```suggestion
            self.prompts = [self.goal] + beast_output
```
or is this supposed to prepend `self.goal` to each output generated by `run_beast`?
```suggestion
            self.prompts = [self.goal + val for val in beast_output]
```

Reply:
Ah yeah, this is an artifact of the change to our generators. Good catch.